src/services/AccountService.js

```diff
@@ -35,72 +35,90 @@ export default class AccountService {
     static getAddressByAccountModelAndNetwork(accountModel: AccountModel, network: AppNetworkType): string {
         return Account.createFromPrivateKey(accountModel.privateKey, this._appNetworkToNetworkType(network)).address.pretty();
     }
+    /**
+     * Generates random mnemonic
+     */
+    static async getNextIndex(network: AppNetworkType): number {
+        const accounts = await AccountSecureStorage.getAllAccountsByNetwork(network);
+        const hdAccounts = accounts.filter(el => el.type === 'hd' && el.path);
+        hdAccounts.sort((a, b) => {
```

Comment:
const nextIndex = Math.max.apply(Math, hdAccounts.map(account => this.getIndexFromAccountModel(account))) + 1;